File: php/program/Lampu.php
<?php
require_once "Elektronik.php";

class Lampu extends Elektronik {
    private $bentuk;
    private $warna;
    private $tipe;
    private $gambar;

    public function __construct($kodeProduk, $nama, $stok, $merk, $daya, $harga, $bentuk, $warna, $tipe, $gambar) {
        parent::__construct($kodeProduk, $nama, $stok, $merk, $daya, $harga);
        $this->bentuk = $bentuk;
        $this->warna = $warna;
        $this->tipe = $tipe;
        $this->gambar = $gambar; // nama file gambar
    }

    public function getData() {
        return [
            "kodeProduk" => $this->kodeProduk,
            "nama" => $this->nama,
            "stok" => $this->stok,
            "merk" => $this->merk,
            "daya" => $this->daya,
            "harga" => $this->harga,
            "bentuk" => $this->bentuk,
            "warna" => $this->warna,
            "tipe" => $this->tipe,
            "gambar" => $this->gambar
        ];
    }
}

File: python/program/lampu.py
from elektronik import Elektronik

class Lampu(Elektronik):
    def __init__(self, kode_produk, nama, stok, merk, daya, harga, bentuk, warna, tipe):
        super().__init__(kode_produk, nama, stok, merk, daya, harga)
        self.bentuk = bentuk
        self.warna = warna
        self.tipe = tipe

    def add(self):
        print(f"Menambahkan Lampu: {self.nama}, Merk: {self.merk}, Harga: {self.harga}")

    @staticmethod
    def create_from_input():
        kode = input("Masukkan kode produk: ")
        nama = input("Masukkan nama produk: ")
        stok = int(input("Masukkan stok: "))
        merk = input("Masukkan merk: ")
        daya = input("Masukkan daya: ")
        harga = float(input("Masukkan harga: "))
        bentuk = input("Masukkan bentuk: ")
        warna = input("Masukkan warna: ")
        tipe = input("Masukkan tipe: ")
        return Lampu(kode, nama, stok, merk, daya, harga, bentuk, warna, tipe)
